sunpy/map/maputils.py

```diff
@@ -511,3 +516,98 @@ def extract_along_coord(smap, coord):
     pixel_coords = smap.wcs.pixel_to_world(pix[:, 1], pix[:, 0])
 
     return pixel_values, pixel_coords
+
+
+def _intersected_pixels(*, x1, y1, x2, y2):
+    """
+    Returns an array of all pixel coordinates which the line defined by `x1, y1` and
+    `x2, y2` crosses.
+
+    Parameters
+    ----------
+    x1, y1, x2, y2 : `float`
+    """
+    dx = x2 - x1
+    dy = y2 - y1
+    dr = np.sqrt(dx ** 2 + dy ** 2)
+
+    # Get the integer pixels for the start and end points
+    ix1, iy1, ix2, iy2 = np.rint([x1, y1, x2, y2]).astype(int)
+
+    # Create the arrays for pixel indices
+    nx = np.abs(ix2 - ix1) + 1
+    ny = np.abs(iy2 - iy1) + 1
+    x, y = np.indices((nx, ny))
+    x = x * np.sign(dx) + ix1
+    y = y * np.sign(dy) + iy1
+
+    # Calculate the distance from the line segment in pixels
+    distance = np.abs(dy * (x - x1) - dx * (y - y1)) / dr
+
+    # The threshold distance is half a pixel times an adjustment for line angle
+    threshold = 0.5 * (np.abs(dx) + np.abs(dy)) / dr
+
+    use = distance <= threshold
+    return np.stack([x[use], y[use]], axis=1)
+
+
+def pixelate_coord_path(smap, coord_path, *, bresenham=False):
+    """
+    Return the pixel coordinates for every pixel that intersects with a coordinate
+    path.
+
+    Each pair of consecutive coordinates in the provided coordinate array defines a
+    line segment.  Each pixel that intersects with a line segment has the
+    coordinates of its center returned in the output.
+
+    To obtain the values of these pixels, pass the output to
+    :func:`~sunpy.map.sample_at_coords`.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        The sunpy map.
+    coord : `~astropy.coordinates.SkyCoord`
+        The coordinate path.
+    bresenham : `bool`
+        If ``True``, use Bresenham's line algorithm instead of the default
```

Comment:
```suggestion
        If `True`, use Bresenham's line algorithm instead of the default
```
We can link it, should we?

Reply:
My inclination is to not link `True` because it adds one more link that someone would need to step through if they are stepping through links, as well as creating a hover, all for little to no actual benefit.